ekundayo-ab/hello-books
# #152364790 Paginate book list on pages with many data

### What does this PR do?
It ensures the pagination logic is extracted into a reusable component and reusable helper function with corresponding server fetching changes. Also, the API test which affects this feature is made to pass

### Description of Task to be completed?
- Pagination should be effected on the server side first from the query made to the database
- On the client side, links and navigation buttons which makes data to be accessed in different pages should be added

### How should this be manually tested?
1. Pull in changes from **remote staging** into **local staging**
2. Start up application with `npm run start:dev` first and then `npm run build:dev`
3. Navigation to url specified in the terminal [http://localhost:3000](http://localhost:3000)
4. Register then Login to the application with details used in registering.
5. Navigate to the **Shelf**, **History**, **Admin Dashboard** and **Profile** page to see the pagination feature

### Any background context you want to provide?
Pagination links are hidden when no data exists on the page 

### What are the relevant pivotal tracker stories?
#152364790 Paginate book list on pages with many data

### Screenshots (if appropriate)
<img width="1436" alt="screen shot 2017-11-13 at 2 23 42 pm" src="https://user-images.githubusercontent.com/20375577/32727818-4e3c69f4-c87e-11e7-9dc3-8f17c12a4fd9.png">

### Questions:
N/A
